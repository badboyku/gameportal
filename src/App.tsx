import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import routes from './routes/index';

const App = () => <RouterProvider router={createBrowserRouter(routes)} />;

export default App;
