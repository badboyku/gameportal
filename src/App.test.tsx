import { render, screen } from '@testing-library/react';
import App from './App';

const renderApp = (props = {}) => {
  return render(<App {...props} />);
};

describe('Component App', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderApp();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the default route "App" -> "Routes -> "HomePage" -> "HelloWorld"', () => {
    renderApp();

    expect(screen.getByText('Hello World!')).toBeInTheDocument();
  });
});
