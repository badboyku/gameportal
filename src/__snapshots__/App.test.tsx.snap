// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component App renders without crashing 1`] = `
<DocumentFragment>
  <div
    style="text-align: center;"
  >
    <h1
      class="error"
    >
      We're sorry, but something went wrong
    </h1>
    <h3>
      You may go back, or click below to try again
    </h3>
    <button
      type="button"
    >
      Retry
    </button>
  </div>
</DocumentFragment>
`;
